Islandora/islandora_scholar
# Islandora 2251 Resubmitted

## Resubmitted from https://github.com/Islandora/islandora_scholar/pull/295 due to an issue with the branch. 

**JIRA Ticket**: https://jira.duraspace.org/browse/ISLANDORA-2251

https://docs.google.com/spreadsheets/d/1PtK16Fo2q4jTMgz17DYmV1nauvoGDEi6f2fhl9K1ZIA

# What does this Pull Request do?

Makes the XPaths editable through an admin GUI

![ISLANDORA SCHOLAR XPATH CONFIGURATION](https://user-images.githubusercontent.com/2738244/43010388-539f4278-8c0e-11e8-89b9-affbe6287451.png)

### Enabled
![Complete](https://user-images.githubusercontent.com/2738244/43728147-28545948-9972-11e8-88b6-82de81703268.png)

# What's new?

Islandora Scholar XPath Configuration Section located on the /admin/islandora/solution_pack_config/scholar page. 

1. It allows you to modify the modules' XPaths.
2. It allows you reset the xpaths back to the original values.
3. It will show in real time preview of the values of a given MODS XPath value for a given PID. Preview returns the value in 1 of 4 states.
     a. Path cannot be blank (User provided XPath is blank and not allowed)
     b. Error (User provided XPath has a syntax error)
     c. Nothing found at the specified XPath
     d. Found value at the specified XPath

# How should this be tested?
* Start islandora_vagrant and ingest an item (prior to switching to this PR)
  * Inspect it and test that the search results look like expects
  * from the terminal
```terminal
      $ vagrant ssh
      $ cd /var/www/drupal
      $ drush vget islandora_scholar_default_search_xpath
      No matching variable found.
```
* Switch to this PR
  * Go to /admin/islandora/solution_pack_config/scholar page and click 'Save Configuration'
  * from the terminal
```terminal
      $ vagrant ssh
      $ cd /var/www/drupal
      $ drush vget islandora_scholar_default_search_xpath
...
```
  * This time you should see several results
  * Ingest an item
  * Try modifying an XPath (easiest is to switch title with something else)
  * Verify the MODS validation is working as expected(give it a valid value, an invalid value, a blank value and possible an error causing value). 
  * Check that the change shows up on a new ingest and an existing one. 
  * Also try adding new values to the citation_form.xml or thesis_form.xml and see if pointing to the new values works. 

# Additional Notes:

* This change requires documentation to be updated. Only placeholders were added. 
* This should not impact existing installations.
* This does **NOT** modify the submission form.
* This allows for the 52 XPaths and 3 subpaths to be set
* This exclude subpaths in a few spots where they occurred.
    * Check google doc for excluded some paths

Using some feedback (from committers call) I set the menu to default to collapsed to keep the screen clean.
![0](https://user-images.githubusercontent.com/2738244/43010444-7b670390-8c0e-11e8-82c4-6c30c0045407.png)

# Interested parties
Tag (@ mention) interested parties or, if unsure, @Islandora/7-x-1-x-committers
